src/Microsoft.Extensions.Http.Polly/PolicyExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+
+namespace Polly
+{
+    public static class PolicyExtensions
+    {
+        public static PolicyBuilder<HttpResponseMessage> HandleTransientHttpError()
+        {
+            return Policy.Handle<HttpRequestException>().OrTransientHttpStatusCode();
+        }
+
+        public static PolicyBuilder<HttpResponseMessage> OrTransientHttpStatusCode(this PolicyBuilder policyBuilder)
+        {
+            if (policyBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(policyBuilder));
+            }
+
+            return policyBuilder.OrResult<HttpResponseMessage>(response =>
+            {
+                return (int)response.StatusCode >= 500 || response.StatusCode == HttpStatusCode.RequestTimeout;
+            });
+        }
+
+        public static PolicyBuilder<HttpResponseMessage> OrTransientHttpError(this PolicyBuilder policyBuilder)
+        {
+            if (policyBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(policyBuilder));
+            }
+
+            return policyBuilder.Or<HttpRequestException>().OrTransientHttpStatusCode();
+        }
+    }
```

Comment:
From where I'm sitting the exceptions and status codes covered by the opinionated methods here feels like a coherent set :+1: (tho intd in other views).  Users can also extend the reach of exceptions or status codes the policy handles with Polly's fluent syntax.

---

We might want to add the following additional two overloads:

```csharp
    public static PolicyBuilder<HttpResponseMessage> OrTransientHttpStatusCode(this PolicyBuilder<HttpResponseMessage> policyBuilder)
        {
            if (policyBuilder == null)
            {
                throw new ArgumentNullException(nameof(policyBuilder));
            }

            return policyBuilder.OrResult<HttpResponseMessage>(response =>
            {
                return (int)response.StatusCode >= 500 || response.StatusCode == HttpStatusCode.RequestTimeout;
            });
        }

        public static PolicyBuilder<HttpResponseMessage> OrTransientHttpError(this PolicyBuilder<HttpResponseMessage> policyBuilder)
        {
            if (policyBuilder == null)
            {
                throw new ArgumentNullException(nameof(policyBuilder));
            }

            return policyBuilder.Or<HttpRequestException>().OrTransientHttpStatusCode();
        }
```

_Explanation:_ If the user starts defining a policy with:

    Policy.HandleResult<HttpResponseMessage>(r => r.StatusCode == /* some other status code they also want to handle */)

then they have a `PolicyBuilder<HttpResponseMessage>` right off the bat.  So, for completeness, the above provides variants of `OrTransientHttpStatusCode(...)` and `OrTransientHttpError(...)` which also chain fluently on `PolicyBuilder<HttpResponseMessage>`.

(guess it would then also be an option to factor out some common code across the overloads)

Reply:
Makes sense.